lib/src/editor/ui/doc_component.dart

```diff
@@ -0,0 +1,152 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_feather_icons/flutter_feather_icons.dart';
+import 'package:flutter_highlight/flutter_highlight.dart';
+import 'package:flutter_highlight/themes/atom-one-dark.dart';
+import 'package:flutterbook/src/editor/ui/copy_text.dart';
+import 'package:flutterbook/src/editor/ui/styled_icon_button.dart';
+import 'package:flutterbook/src/utils/extensions.dart';
+import 'package:flutterbook/src/utils/radii.dart';
+
+class DocPanel extends StatefulWidget {
+  final Widget component;
+  final String? docs;
+  final String stateName;
+
+  const DocPanel({
+    Key? key,
+    required this.component,
+    this.docs,
+    required this.stateName,
+  }) : super(key: key);
+
+  @override
+  _DocPanelState createState() => _DocPanelState();
+}
+
+class _DocPanelState extends State<DocPanel> {
+  bool expanded = false;
+  double _zoom = 1;
+  get zoom => _zoom;
+
+  void zoomIn() {
+    setState(() {
+      _zoom += 0.25;
+    });
+  }
+
+  void zoomOut() {
+    setState(() {
+      _zoom = (_zoom - 0.25).clamp(0.5, 999);
+    });
+  }
+
+  void resetZoom() {
+    setState(() {
+      _zoom = 1;
+    });
+  }
+
+  void toggleExpansion() {
+    setState(() {
+      expanded = !expanded;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final TextStyle tabStyle = context.textTheme.subtitle1!.copyWith(
```

Comment:
Is `context.textTheme` getting us the same thing as `Theme.of(context).textTheme`? I can't remember if it traces up the tree this way.

Reply:
it worked !